ui/src/vertical-geometry/vertical-geometry-diagram.tsx

```diff
@@ -59,6 +59,26 @@ interface VerticalGeometryDiagramProps {
     showArea: (area: BoundingBox) => void;
 }
 
+function loadGeometry(
+    changeTimes: ChangeTimes,
+    alignmentId:
+        | { planId: GeometryPlanId; alignmentId: GeometryAlignmentId }
+        | { locationTrackId: LocationTrackId; publishType: PublishType },
+): Promise<VerticalGeometryItem[] | null | undefined> {
+    return 'planId' in alignmentId
```

Comment:
Mites tämä vertailu toimii? En ole aiemmin nähnyt että `in`-operaattoria käytettäisiin siten, että oikealla puolella on string

Reply:
Se katsoo, onko oliossa kyseisen niminen kenttä. Kätevä siksi, että typescript osaa käyttää sitä tyyppitietona, ts. jos alignmentId:ssä on tässä planId-kenttä, niin typescript tietää sen olevan tuo geometriatyypin haara.

Reply:
Mutta eikös `alignmentId`:n pitäisi olla alias `string`:ille? Ei kai siellä pitäisi mitään programmer-defined-kenttiä olla? 🤔 Tai siis, olisihan se plain JavaScriptissä ihan mahdollista prototyyppi- tms. kikkailulla, mutta en muista nähneeni meillä tuollaista, enkä tiedä mitä TypeScript olisi asiasta jos sellaista yrittäisi.